source/docs/software/advanced-gradlerio/compiler-args.rst

```diff
@@ -0,0 +1,45 @@
+Using Compiler Arguments
+============================
+
+Compiler arguments allow us to change the behavior of our compiler. This includes making warnings into errors, ignoring certain warnings and choosing optimization level. When you compile your code a variety of flags are already included by default which can be found `here <https://github.com/wpilibsuite/native-utils/blob/0055bc77da9d9d37e0876d04af53b7317b78174f/src/main/java/edu/wpi/first/nativeutils/WPINativeUtilsExtension.java#L37>`. Normally we would pass them in as flags when compiling our code but this doesn't work in GradleRIO. Instead we have to modify our build.gradle.
+
+C++
+---
+
+Platforms
+^^^^^^^^^
+
+Different compilers on different platforms use a variety of different flags so in order to avoid breaking different platforms with compiler flags we configure everything per platform. The platforms that are supported are :
+
+- windowsx86-64
+- windowsx86
+- windowsarm64
+- linuxathena (roboRIO)
+- linuxarm32
```

Comment:
I think there is a gradle file we can link instead of duplicating here. @ThadHouse

Reply:
I got that from looking through native utils. I can try and find the file where they were listed

Reply:
found [here](https://github.com/wpilibsuite/native-utils/blob/5b6aca5efed38188561ec26b58bebbe159e5f340/src/main/java/edu/wpi/first/nativeutils/WPINativeUtilsExtension.java#L97)

Reply:
Should I link this instead @Daltz333